src/com/franz/sud/components/engine/GameManager.java

```diff
@@ -109,6 +110,7 @@ private void getUserAction() {
      * it is not listed in the open rooms of the current map used.
      */
     private void move() {

```

Comment:
Instead of recursion, convert this to a loop and remove calls of `move()` in `moveTo()`